22_pipeslab/pipeslab.txt

```diff
@@ -2,20 +2,44 @@ For each of the following, type a single command line that performs:
 
   1. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner.
   Answer: 
+	ls -l|grep $USER| cut -c 4|grep x|wc -w
+	[Q: Based on the commands we know by now is there a shorter way to do 	this? With less pipes?] 
 
   2. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner, group and others.
   Answer: 
+	ls -l|grep $USER| cut -c 4,7,10|grep x|wc -w
 
   3. Print the number of lines from "ls -l" which represent paths that were modified today (no matter which day we run the command)
   Answer: 
+	
+	ls -l | cut -f6-8 -d" "| grep $(date +%b" "%d)| wc -l
+	
+	[Note: this answer is not working because date +%d return an answer for 	today day as 07 (two char) and in ls print it's only 7(one char)
+	I could check with or and add the “0” char if need but I think you are 	looking for a different answer?
+	Just in case when checking with the string “Jun 07” it's working but I 	know it's not what we are looking for,
+	Also I had a problem when cutting date from folders, the colums where 	different with the date comapared to file? I am not sure for the reason? 
+
+	Below an answer where I used a set date for today:
+	ls -l | cut -f6-8 -d" "| grep "Jun  7"| wc -l (working)
+	]
 
   4. Modify (1) and (2) to count files only (no directories)
   Answer:
 
+	(1) ls -l|grep $USER|cut -c 1-4|grep x |cut -c1 | grep - | wc -l
+	(2) ls -l|grep $USER|cut -c 1,4,7,10|grep x| cut -c1 | grep -|wc -w
+
+
   5. The command /sbin/ifconfig shows information regarding the machine's network interfaces. Use the command to print only the IP address of the machine. Remember not to show the special loopback address 127.0.0.1
 
+	/sbin/ifconfig | tr -s " " | cut -f3 -d" " | grep addr | head -1
```

Comment:
This will only print the first line, while in real life there can be more, and order is not always known. Luckily we can use `grep -v` to say "print everything without the word" getting: `/sbin/ifconfig | ... | grep -v 127.0.0.1`